include/circt/Dialect/RTL/RTLStatements.td

```diff
@@ -91,3 +91,22 @@ def ArrayIndexOp
     $input`[`$index`]` attr-dict `:` type($input) `,` type($index)
   }];
 }
+
+def StructExplodeOp : RTLOp<"struct_explode", [NoSideEffect]> {
+  let summary = "Expand a struct into its constituent parts.";
+  let arguments = (ins StructType:$input);
+  let results = (outs Variadic<RTLValueType>:$result);
+
+  let parser = [{ return ::parse$cppClass(parser, result); }];
+  let printer = [{ return ::print(p, *this); }];
+}
+
+
+def StructCreateOp : RTLOp<"struct_create", [NoSideEffect]> {
+  let summary = "Create a struct from constituent parts.";
+  let arguments = (ins Variadic<RTLValueType>:$input);
+  let results = (outs StructType:$result);
+
+  let parser = [{ return ::parse$cppClass(parser, result); }];
+  let printer = [{ return ::print(p, *this); }];
```

Comment:
Curiosity: Is the only reason that you're using a custom parser and printer here is to make the result type get inferred from the syntax?

Reply:
Yes, I was trying to get a TypesMatchWith to work, but was running into issues with variadics.

Reply:
Yes,this addition will be a huge thing for making MLIR more useful for us in general.